src/queryParameters/extractCreatedByParameters.ts

```diff
@@ -2,35 +2,57 @@ import { ajv } from '../ajv';
 import { InputValidationError } from '../errors';
 import type { JSONSchemaType } from 'ajv';
 import type { Request } from 'express';
+import type { AuthContext } from '../types';
+
+interface CreatedByQueryParameters {
+	createdBy: number | 'me' | undefined;
+}
 
 interface CreatedByParameters {
 	createdBy: number | undefined;
 }
 
-const createdByParametersSchema: JSONSchemaType<CreatedByParameters> = {
-	type: 'object',
-	properties: {
-		createdBy: {
-			type: 'integer',
-			minimum: 1,
-			nullable: true,
+const createdByQueryParametersSchema: JSONSchemaType<CreatedByQueryParameters> =
+	{
+		type: 'object',
+		properties: {
+			createdBy: {
+				type: ['integer', 'string'],
+				oneOf: [
+					{
+						type: 'integer',
+						minimum: 1,
+						nullable: true,
+					},
+					{
+						enum: ['me'],
+					},
+				],
+				nullable: true,
+			},
 		},
-	},
-	required: [],
-};
+		required: [],
+	};
 
-const isCreatedByParameters = ajv.compile(createdByParametersSchema);
+const isCreatedByQueryParameters = ajv.compile(createdByQueryParametersSchema);
 
-const extractCreatedByParameters = (request: Request): CreatedByParameters => {
-	const { query } = request;
-	if (!isCreatedByParameters(query)) {
+const extractCreatedByParameters = (
+	partialRequest: Pick<Request, 'query'>,
+	partialAuthContext?: Pick<AuthContext, 'user'>,
+): CreatedByParameters => {
+	const { query } = partialRequest;
+	if (!isCreatedByQueryParameters(query)) {
 		throw new InputValidationError(
 			'Invalid createdBy parameters.',
-			isCreatedByParameters.errors ?? [],
+			isCreatedByQueryParameters.errors ?? [],
 		);
 	}
+
+	const createdBy =
+		query.createdBy === 'me' ? partialAuthContext?.user.id : query.createdBy;
+
 	return {
-		createdBy: query.createdBy,
+		createdBy,
```

Comment:
this is legitimately a more elegant syntax

Reply:
OK Justin. But listen. Isn't it kind of amazing that we expanded the functionality of TWO endpoints without even editing EITHER of them!?

Reply:
Yes!